### DISCUSSION TITLE
GNN CheatSheet

### DISCUSSION THREAD

**Discussion by nkamalesh111** [pyg-team/pytorch_geometric#8136] [Q&A]
Hello everyone, I am new to this field. I am going through the documentation and there I found this : GNN Cheatsheet: https://pytorch-geometric.readthedocs.io/en/latest/cheatsheet/gnn_cheatsheet.html. Please explain what is the purpose of this as it shows a table and there are tick marks. Please explain how to understand the first function with the related attributes and then I can proceed further.Thank you. Regards Kamalesh.

**Comment by wsad1** [comment]
Each tick indicates that the GNN supports that features. For example take `GCNConv`, 
1. it supports passing message passing based on `SparseTensor`, which is a efficient format to represent edges
2. it supports having weight on the edges so `edge_weight` is ticked
3. but its doesn't support edge attributes i.e multi-deimensional features on the edge so `edge_attr` is not ticked, 
4. bipartite graphs are not supported as input. Read more about such graphs [here](https://pytorch-geometric.readthedocs.io/en/latest/advanced/batching.html#bipartite-graphs)
5. lazy is ticked, as you don't need to define the number of input features while initializing `GCNConv` it will autoinfer the number of inputs when you first call forward.
